File: src/Implicit/ImplicitBlend.java
package Implicit;

import util.Maths;

public final class ImplicitBlend extends ImplicitModuleBase {
    public ImplicitBlend(final ImplicitModuleBase source, final double low, final double high) {
        this.Source = source;
        this.Low = new ImplicitConstant(low);
        this.High = new ImplicitConstant(high);
    }
    
    private ImplicitModuleBase Source;
    
    private ImplicitModuleBase Low;
    
    private ImplicitModuleBase High;
    
    public ImplicitModuleBase getSource() {
        return this.Source;
    }
    
    public void setSource(final ImplicitModuleBase source) {
        this.Source = source;
    }
    
    public ImplicitModuleBase getLow() {
        return this.Low;
    }
    
    public void setLow(final ImplicitModuleBase low) {
        this.Low = low;
    }
    
    public ImplicitModuleBase getHigh() {
        return this.High;
    }
    
    public void setHigh(final ImplicitModuleBase high) {
        this.High = high;
    }
    
    @Override
    public double Get(final double x, final double y) {
        final double v1 = this.Low.Get(x, y);
        final double v2 = this.High.Get(x, y);
        final double blend = (this.Source.Get(x, y) + 1.0) * 0.5;
        return Maths.Lerp(blend, v1, v2);
    }
    
    @Override
    public double Get(final double x, final double y, final double z) {
        final double v1 = this.Low.Get(x, y, z);
        final double v2 = this.High.Get(x, y, z);
        final double blend = this.Source.Get(x, y, z);
        return Maths.Lerp(blend, v1, v2);
    }
    
    @Override
    public double Get(final double x, final double y, final double z, final double w) {
        final double v1 = this.Low.Get(x, y, z, w);
        final double v2 = this.High.Get(x, y, z, w);
        final double blend = this.Source.Get(x, y, z, w);
        return Maths.Lerp(blend, v1, v2);
    }
    
    @Override
    public double Get(final double x, final double y, final double z, final double w, final double u, final double v) {
        final double v1 = this.Low.Get(x, y, z, w, u, v);
        final double v2 = this.High.Get(x, y, z, w, u, v);
        final double blend = this.Source.Get(x, y, z, w, u, v);
        return Maths.Lerp(blend, v1, v2);
    }
}

File: src/Implicit/ImplicitBlend.cs
using System;

namespace TinkerWorX.AccidentalNoiseLibrary
{
    public sealed class ImplicitBlend : ImplicitModuleBase
    {
        public ImplicitBlend(ImplicitModuleBase source, double low = 0.00, double high = 0.00)
        {
            this.Source = source;
            this.Low = new ImplicitConstant(low);
            this.High = new ImplicitConstant(high);
        }

        public ImplicitModuleBase Source { get; set; }

        public ImplicitModuleBase Low { get; set; }

        public ImplicitModuleBase High { get; set; }

        public override double Get(double x, double y)
        {
            var v1 = this.Low.Get(x, y);
            var v2 = this.High.Get(x, y);
            var blend = (this.Source.Get(x, y) + 1.0) * 0.5;
            return Utilities.Maths.Lerp(blend, v1, v2);
        }

        public override double Get(double x, double y, double z)
        {
            var v1 = this.Low.Get(x, y, z);
            var v2 = this.High.Get(x, y, z);
            var blend = this.Source.Get(x, y, z);
            return Utilities.Maths.Lerp(blend, v1, v2);
        }

        public override double Get(double x, double y, double z, double w)
        {
            var v1 = this.Low.Get(x, y, z, w);
            var v2 = this.High.Get(x, y, z, w);
            var blend = this.Source.Get(x, y, z, w);
            return Utilities.Maths.Lerp(blend, v1, v2);
        }

        public override double Get(double x, double y, double z, double w, double u, double v)
        {
            var v1 = this.Low.Get(x, y, z, w, u, v);
            var v2 = this.High.Get(x, y, z, w, u, v);
            var blend = this.Source.Get(x, y, z, w, u, v);
            return Utilities.Maths.Lerp(blend, v1, v2);
        }
    }
}
